tool/src/js/mod.rs

```diff
@@ -29,6 +29,16 @@ pub fn gen_bindings(
     outs.entry("diplomat-runtime.mjs".to_string())
         .or_default()
         .write_str(include_str!("../../templates/js2/runtime.mjs"))?;
+    outs.get_mut("diplomat-runtime.mjs")
+        .unwrap()
+        .write_str(concat!(
+            "export class FFIError extends Error {\n",
```

Comment:
We eventually want to mark types as errors instead, yes? https://github.com/rust-diplomat/diplomat/issues/446

Reply:
I'm not sure, the `cause` solution seems nice enough. TS won't be able to strongly type the errors anyway.